src/hooks/usePoolsHandler.tsx

```diff
@@ -1,21 +1,53 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useCallback } from 'react'
 import { notifyError } from './useNotification'
-import { PoolsQueryDocument, PoolsQueryQuery, execute } from '../../.graphclient'
+import { ExecutionResult } from 'graphql'
+import { PoolsDocument, Pool, PoolsQuery, LatestPoolDataDocument, execute, PoolData } from '../../.graphclient'
 
 export const usePoolsHandler = () => {
-  const [poolsData, setPoolsData] = useState<PoolsQueryQuery>()
+  const [pools, setPools] = useState<ExecutionResult<PoolsQuery>>()
+  const [latestPoolsData, setLatestPoolsData]  = useState<Array<PoolData>>([])
   
-  useEffect(() => {
-    const fetchPoolsData = async () => {
-      const res = await execute(PoolsQueryDocument, {})
-      if (res?.data) {
-        setPoolsData(res?.data)
-      } else {
-        console.log("NO RESPONSE")
+  // fetches all pool entities
+  const fetchPoolsData = useCallback(async () => {
+    const res = await execute(PoolsDocument, {})
+    if (res?.data) {
+      setPools(res.data.pools)
+    } else {
+      console.log("NO RESPONSE")
+    }
+  }, [])
+  
+  // fetches pool's latest data by pool address
+  const fetchLatestPoolData = async (address: string): Promise<PoolData | number> => {
+    const res = await execute(LatestPoolDataDocument, { address })
+    if (res?.data.poolDatas[0]) {
+      return res.data.poolDatas[0]
+    }
+  
+    return 0
+  }
+  
+  // iterates through all pool addresses and fetches latest pool data for it
+  const fetchLatestPoolsData = useCallback(async (pools: Array<Pool>) => {
+    const newPoolsData: Array<PoolData> = []
+
+    for (const pool of pools) {
+      const latestPoolData: PoolData | number = await fetchLatestPoolData(pool.address)
+      if (latestPoolData != 0) {
+        newPoolsData.push(latestPoolData as PoolData)
       }
     }
-    fetchPoolsData()
+    
+    setLatestPoolsData(newPoolsData)
   }, [])
+  
+  useEffect(() => {
+    fetchPoolsData()
+  }, [fetchPoolsData])
```

Comment:
Can this be called on page load like:
useEffect(() => {
    fetchPoolsData()
  }, [])

Reply:
<img width="398" alt="Screen Shot 2022-10-19 at 7 43 56 AM" src="https://user-images.githubusercontent.com/63836051/196681610-fb0616bf-dce4-4dbd-9a2f-3b99352d3d0e.png">

so rendering it just once makes double the renders as opposed to rendering it per state change of `fetchPoolsData`